File: app/src/main/java/com/calculatemypace/david/calculateMyPace/StopwatchActivity.java
package com.calculatemypace.david.calculateMyPace;

import android.os.Handler;
import android.os.SystemClock;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.view.View.OnClickListener;
import android.widget.TextView;

public class StopwatchActivity extends AppCompatActivity {

    Button butnstart, butnreset;
    TextView time;
    long starttime = 0L;
    long timeInMilliseconds = 0L;
    long timeSwapBuff = 0L;
    long updatedtime = 0L;
    int t = 1;
    int secs = 0;
    int mins = 0;
    int milliseconds = 0;
    Handler handler = new Handler();


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_stopwatch);
        butnstart = (Button) findViewById(R.id.start_button);
        butnreset = (Button) findViewById(R.id.reset_button);
        time = (TextView) findViewById(R.id.timer);
        time.setText("0:00:00");


        butnstart.setOnClickListener(new OnClickListener() {

            @Override
            public void onClick(View v) {

                if (t == 1) {
                    butnstart.setText("Pause");
                    starttime = SystemClock.uptimeMillis();
                    handler.postDelayed(updateTimer, 0);
                    t = 0;
                } else {
                    butnstart.setText("Start");
                    timeSwapBuff += timeInMilliseconds;
                    handler.removeCallbacks(updateTimer);
                    t = 1;
                }
            }

        });

        butnreset.setOnClickListener(new OnClickListener() {

            @Override
            public void onClick(View v) {

                starttime = 0L;
                timeInMilliseconds = 0L;
                timeSwapBuff = 0L;
                updatedtime = 0L;
                t = 1;
                secs = 0;
                mins = 0;
                milliseconds = 0;
                butnstart.setText("Start");
                handler.removeCallbacks(updateTimer);
                time.setText("0:00:00");
            }});

        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setHomeButtonEnabled(true);


    }

    public Runnable updateTimer = new Runnable() {
        public void run() {
            timeInMilliseconds = SystemClock.uptimeMillis() - starttime;
            updatedtime = timeSwapBuff + timeInMilliseconds;
            secs = (int) (updatedtime / 1000);
            mins = secs / 60;
            secs = secs % 60;
            milliseconds = (int) (updatedtime % 1000);
            time.setText("" + mins + ":" + String.format("%02d", secs) + ":" + String.format("%03d", milliseconds));
            handler.postDelayed(this, 0);
        }};


}

File: app/src/main/java/com/calculatemypace/david/calculatemypace/StopwatchActivity.kt
package com.calculatemypace.david.calculatemypace

import android.app.Activity
import android.widget.TextView
import android.os.Bundle
import android.os.Handler
import android.os.SystemClock
import android.view.View
import android.widget.Button
import butterknife.BindView
import butterknife.ButterKnife
import org.w3c.dom.Text

class StopwatchActivity : Activity() {
    @BindView(R.id.start_button) lateinit var buttonStart: Button
    @BindView(R.id.reset_button) lateinit var buttonReset: Button
    @BindView(R.id.timer) lateinit var time: TextView

    var startTime = 0L
    var timeInMilliseconds = 0L
    var timeSwapBuff = 0L
    var updatedTime = 0L
    var t = 1
    var secs = 0
    var mins = 0
    var milliseconds = 0
    var handler = Handler()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_stopwatch)
        ButterKnife.bind(this)

        time.text = getString(R.string.start_time)

        buttonStart.setOnClickListener {
            startWatch()
        }
        buttonReset.setOnClickListener {
            reset()
        }
    }

    private var updateTimer: Runnable = object : Runnable {
        override fun run() {
            timeInMilliseconds = SystemClock.uptimeMillis() - startTime
            updatedTime = timeSwapBuff + timeInMilliseconds
            secs = (updatedTime / 1000).toInt()
            mins = secs / 60
            secs = secs % 60
            milliseconds = (updatedTime % 1000).toInt()
            time.text = "" + mins + ":" + String.format("%02d", secs) + ":" + String.format(
                "%03d",
                milliseconds
            )
            handler.postDelayed(this, 0)
        }
    }

    private fun startWatch() {
        if (t == 1) {
            buttonStart.text = getString(R.string.pause_button)
            startTime = SystemClock.uptimeMillis()
            handler.postDelayed(updateTimer, 0)
            t = 0
        } else {
            buttonStart.text = getString(R.string.start_button)
            timeSwapBuff += timeInMilliseconds
            handler.removeCallbacks(updateTimer)
            t = 1
        }
    }

    private fun reset() {
        startTime = 0L
        timeInMilliseconds = 0L
        timeSwapBuff = 0L
        updatedTime = 0L
        t = 1
        secs = 0
        mins = 0
        milliseconds = 0
        buttonStart.text = getString(R.string.start_button)
        handler.removeCallbacks(updateTimer)
        time.text = getString(R.string.start_time)
    }
}